karabo_data/geometry2.py

```diff
@@ -631,6 +737,228 @@ def plot_data(self, modules_data):
+          quadrant.
+        unit : float, optional
+          The conversion factor to put the coordinates into metres.
+          The default 1e-3 means the numbers are in millimetres.
+        """
+        assert len(positions) == 4
+        modules = []
+        with h5py.File(path, 'r') as f:
+            for Q, M in product(range(1, 5), range(1, 5)):
+                quad_pos = np.array(positions[Q - 1])
+                mod_grp = f['Q{}/M{}'.format(Q, M)]
+                mod_offset = mod_grp['Position'][:2]
+
+                tiles = []
+                for T in range(1, 17):
+                    corner_pos = np.zeros(3)
+                    tile_offset = mod_grp['T{:02}/Position'.format(T)][:2]
+                    corner_pos[:2] = quad_pos + mod_offset + tile_offset
+
+                    # Convert units (mm) to pixels
+                    corner_pos *= (unit / cls.pixel_size)
+
+                    # LPD geometry is measured to the last pixel of each tile.
+                    # Subtract tile dimensions for the position of 1st pixel.
+                    ss_vec, fs_vec = np.array([0, 1, 0]), np.array([1, 0, 0])
+                    first_px_pos = (corner_pos
+                                    - (ss_vec * cls.frag_ss_pixels)
+                                    - (fs_vec * cls.frag_fs_pixels))
+
+                    tiles.append(GeometryFragment(
+                        corner_pos=first_px_pos,
+                        ss_vec=ss_vec,
+                        fs_vec=fs_vec,
+                        ss_pixels=cls.frag_ss_pixels,
+                        fs_pixels=cls.frag_fs_pixels,
+                    ))
+                modules.append(tiles)
+
+        return cls(modules, filename=path)
+
+
+    def inspect(self, frontview=True):
+        """Plot the 2D layout of this detector geometry.
+
+        Returns a matplotlib Figure object.
+
+        Parameters
+        ----------
+
+        frontview : bool
+          If True (the default), x increases to the left, as if you were looking
+          along the beam. False gives a 'looking into the beam' view.
+        """
+        fig = super().inspect(frontview=frontview)
+        ax = fig.axes[0]
+
+        # Label modules and tiles
+        for ch, module in enumerate(self.modules):
+            s = 'Q{Q}M{M}'.format(Q=(ch // 4) + 1, M=(ch % 4) + 1)
+            cx, cy, _ = module[0].centre()
+            ax.text(cx, cy, s, fontweight='bold',
+                    verticalalignment='center',
+                    horizontalalignment='center')
+
+            for t in [7, 8, 15]:
+                cx, cy, _ = module[t].centre()
+                ax.text(cx, cy, 'T{}'.format(t + 1),
+                        verticalalignment='center',
+                        horizontalalignment='center')
+
+        ax.set_title('LPD-1M detector geometry ({})'.format(self.filename))
+        return fig
+
+    @staticmethod
+    def split_tiles(module_data):
+        lhs, rhs = np.split(module_data, 2, axis=-1)
```

Comment:
As we've had discussed left and right are kind of arbitrary. Nautical terms port and starboard might also be confusing